backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/utils/ApplicationJsonExtensions.kt

```diff
@@ -0,0 +1,36 @@
+package app.ehrenamtskarte.backend.application.webservice.utils
+
+import app.ehrenamtskarte.backend.application.database.ApplicationEntity
+import app.ehrenamtskarte.backend.common.utils.findValueByName
+import app.ehrenamtskarte.backend.common.utils.findValueByPath
+import app.ehrenamtskarte.backend.exception.webservice.exceptions.ApplicationDataIncompleteException
+import com.fasterxml.jackson.databind.JsonNode
+
+object ApplicationJsonExtensions {
+    fun ApplicationEntity.getApplicantFirstName(): String =
+        this.getPersonalDataNode().findValueByName("forenames")
+            ?: throw ApplicationDataIncompleteException()
+
+    fun ApplicationEntity.getApplicantLastName(): String =
+        this.getPersonalDataNode().findValueByName("surname")
+            ?: throw ApplicationDataIncompleteException()
+
+    fun ApplicationEntity.getApplicantDateOfBirth(): String =
+        this.getPersonalDataNode().findValueByName("dateOfBirth")
+            ?: throw ApplicationDataIncompleteException()
+
+    fun ApplicationEntity.getApplicantName(): String {
+        val forenames = this.getApplicantFirstName()
+        val surname = this.getApplicantLastName()
+
+        return listOfNotNull(forenames, surname).filter { it.isNotBlank() }.joinToString(" ")
+    }
+
+    fun ApplicationEntity.getApplicantEmail(): String =
+        this.getPersonalDataNode().findValueByName("emailAddress")
+            ?: throw ApplicationDataIncompleteException()
+
+    fun ApplicationEntity.getPersonalDataNode(): JsonNode =
+        this.parseJsonValue().findValueByPath("application", "personalData")
+            ?: throw ApplicationDataIncompleteException()
```

Comment:
🔧  Since we are using these methods in a variety of packages, they could just as well be defined in the `ApplicationEntity` themselves, preferably as attribute getters, instead of Java-style `get…()` methods.

Also, putting extension methods in an enclosing scope only makes sense, if you want to force the usage sites to have a specific `this` scope, but this would be better solved with [context parameters](https://kotlinlang.org/docs/whatsnew22.html#preview-of-context-parameters), and is not necessary here.

Reply:
> Since we are using these methods in a variety of packages, they could just as well be defined in the ApplicationEntity themselves

I see these functions as utilities to parse the application json.
I don't think applicantName, applicantEmail, etc are ApplicationEntity properties.
I think putting it all in ApplicationEntity will make a clutter there, so I'd rather keep it separate, considering that we might want to parse even more json fields in the future.

Reply:
I can see your point, the might as well be extension methods and we keep the entity classes small.

But the surrounding `object ApplicationJsonExtensions` is unnecessary. In Kotlin freestanding functions (and objects) can exist and are idiomatic. I would even say that `object` is rarely necessary, and is in most cases a Java-ism.

Reply:
Removed ApplicationJsonExtensions object